go/vt/vttablet/tabletserver/planbuilder/dml.go

```diff
@@ -497,16 +497,6 @@ func analyzeInsertMessage(ins *sqlparser.Insert, plan *Plan, table *schema.Table
 	if _, ok := ins.Rows.(sqlparser.SelectStatement); ok {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "subquery not allowed for message table: %s", table.Name.String())
 	}
-	if ins.OnDup != nil {
-		// only allow 'on duplicate key' where time_scheduled and id are not referenced
-		ts := sqlparser.NewColIdent("time_scheduled")
-		id := sqlparser.NewColIdent("id")
-		for _, updateExpr := range ins.OnDup {
-			if updateExpr.Name.Name.Equal(ts) || updateExpr.Name.Name.Equal(id) {
-				return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "'on duplicate key' cannot reference time_scheduled or id for message table: %s", table.Name.String())
-			}
-		}
-	}
```

Comment:
Does this mean that `time_scheduled` and `id` can be updated now? I apologize if my lack of understanding of how vstream functions means that this check is happening somewhere else.

Reply:
Is that something that we should plan on deprecating? Would that allow users to `SELECT ... INSERT INTO message_table`? I've got some breaking messages changes I'd like to contribute and I think adoption is low enough that we can make some long-term changes for the better.

Reply:
Right now, `time_created` and `time_scheduled` are informational. So, we could drop those columns. We could set default values for the other columns. The burden of choosing the correct value for `time_next` will be on the app.

If the above can work, I can make all message inserts pure pass-through.

Reply:
If we got rid of `time_scheduled`, then `time_next` would need to be set to handle scheduling an initial task in the future, which seems reasonable. Otherwise, I think we could set `time_next DEFAULT 1` which would result in it being immediately scheduled, which I would argue is expected behavior.

For the `id` field, we set it explicitly 50% of the time with a FK to another id table, and the other 50% of the time we use a sequence for an auto generated id. Either way works fine with this scenario.

For `time_created`, I wish MySQL would let you default it like you can with a timestamp field. We actually only allow timestamps in our db for created/modified fields for that reason. There's nothing to prevent users from adding their own created field as a timestamp if they want, so I think we're fine eliminating it from the required list.

I think we can probably get any other BC related changes (see https://github.com/vitessio/vitess/issues/5882) merged into this release cycle so there is just one release with breaking changes.